cmd/amppkg/main.go

```diff
@@ -147,11 +164,19 @@ func main() {
 		addr = "localhost"
 	}
 	addr += fmt.Sprint(":", config.Port)
+	var handler http.Handler
```

Comment:
I'd dedupe with:

```
handler := logIntercept{mux}
if config.ProjectId != "" {
  handler = allowProject(config.ProjectId, handler)
  log.Printf(...)
}
```

Reply:
Thanks! I did try something like that originally but couldn't get it working. But, turns out that it needs to some type conversion to address the embedding:

```go
handler := http.Handler(logIntercept{mux})
...
```